tests/test_caching.py

```diff
@@ -831,10 +831,13 @@ def test_solve(self, operate_on_empty_cache):
         # created by the finite difference (u.dt, u.dx2). We would have had
         # three extra references to u(t + dt), u(x - h_x) and u(x + h_x).
         # But this is not the case anymore!
+        # TODO: This cache doesn't clear currently as retained legacy SubDomain API
```

Comment:
This sounds quite dangerous as it might end up preventing Function and TimeFucntion to be reachable by the gc